src/protocol.rs

```diff
@@ -38,73 +27,44 @@ use tracing::{error, info, instrument, trace};
 // Participant API //
 /////////////////////
 
+#[derive(Debug)]
+pub enum ProtocolType {
```

Comment:
Missing documentation.

Reply:
Updated. I'm not quite sold on this type right now, since the caller doesn't actually have to explicitly interact with it -- it's automatically instantiated based on what `ProtocolParticipant` type the caller uses to make a `Participant`. But I think we should leave it for now and just keep it in mind for potential refactoring once we've seen the example and so on.